frontend/src/app/shared/components/date/date.component.ts

```diff
@@ -0,0 +1,36 @@
+import { CommonModule } from "@angular/common";
+import { Component, Input, OnChanges } from "@angular/core";
+import { MatIconModule } from "@angular/material/icon";
+
+@Component({
+    selector: "app-date",
+    standalone: true,
+    imports: [CommonModule, MatIconModule],
+    templateUrl: "./date.component.html",
+    styleUrls: ["./date.component.scss"],
+})
+export class DateComponent implements OnChanges {
+    @Input() date: string;
+    cssClass: string;
+
+    ngOnChanges(): void {
+        const dateObject = new Date(this.date);
+        const today = new Date();
+
+        if (this.areDatesEqual(dateObject, today)) {
```

Comment:
Coś w tym stylu, żeby uniknąć niepotrzebnego chaina dzięki szybszym returnom:?
      
   this.cssClass = this.cssClassPicker(dateObject, today);
   
    private cssClassPicker(inputDate: Date, currentDate: Date): string {
        if (this.areDatesEqual(inputDate, currentDate)) {
            return "date--today";
        } else if (inputDate < currentDate) {
            return "date--past";
        }
        return "date--future";
    }